### PR TITLE
Update setup instructions with newcomers in mind

### PR DESCRIPTION
**Pull request by thibaudcolas**
TL;DR; We could make it easier for people to get involved with Wagtail development by having better (more) instructions on how to do so. This is a WIP, I've asked people on Springload's front-end team to review it at https://github.com/springload/vagrant-wagtail-develop/pull/1. Creating the PR because I'd like to hear other people's thoughts on the matter.

---

I am trying to get more people from Springload and more people in general to contribute to Wagtail, and one of the barriers to entry is properly setting up the project for local development. 

Considering there already is this `vagrant-wagtail-develop` project, and contribution documentation at http://docs.wagtail.io/en/v1.6.3/contributing/index.html, I think it could just be a matter of improving those existing resources instead of making something else.

The way I see it, the following should be addressed:
- How do you get Wagtail up and running locally, with what tools (`vagrant-wagtail-develop`) – that's what this PR covers.
- How are you supposed to make changes (tested, up to Wagtail's standard of quality / styleguides)
- How are you supposed to propose those changes & collaborate (open-source contribution guidelines)

What do you think?

### CODE REVIEW

**Comment by chimeno** [comment]
we should change the remote to wagtail from ``torchbox/wagtail`` to ``wagtail/wagtail``

**Comment by thibaudcolas** [comment]
I think this is good to go – feedback welcome!

File: README.md [hunk 1]
```diff
@@ -13,46 +13,133 @@ Features
...
+# And to switch back to Python 3,
+source ~/.virtualenvs/wagtaildemo/bin/activate
+```
+
+Getting ready to contribute
+---------------------------
+
+Here are other actions you will likely need to do to make your first contribution to the project.
+
+Set up git remotes to Wagtail forks:
```
Comment (gasman) [inline README.md]: We should make it clear that this needs to be done outside of the VM (or else you need to fiddle with SSH keys and things).

File: README.md [hunk 2]
```diff
@@ -13,46 +13,133 @@ Features
...
+# Add springload/wagtail as a "springload" remote.
+git remote add springload git@github.com:springload/wagtail.git
+# Add gasman/wagtail as a "gasman" remote.
+git remote add gasman git@github.com:gasman/wagtail.git
+# Pull latest changes from all remotes / forks.
+git pull --all
+```
+
+Get your development environment configured:
+
```
Comment (gasman) [inline README.md]: Is this referring to IDEs? I think we're better leaving this out - advocating one editor or another really is opening a can of worms :-)

File: README.md [hunk 3]
```diff
@@ -13,46 +13,133 @@ Features
...
+Here are other actions you will likely need to do to make your first contribution to the project.
+
+Set up git remotes to Wagtail forks:
+
+```sh
+cd ~/Development/wagtail/wagtail
+# Change the default origin remote to point to your fork.
+git remote set-url origin https://github.com/<USERNAME>/wagtail.git
+# Add wagtail/wagtail as a "torchbox" remote.
+git remote add torchbox git@github.com:wagtail/wagtail.git
```
Comment (kaedroho) [inline README.md]: Maybe we should rename this remote to something more generic like ``upstream``?

File: README.md [hunk 4]
```diff
@@ -13,46 +13,133 @@ Features
...
-    vagrant ssh
+**Requirements:** [VirtualBox](https://www.virtualbox.org/) and [Vagrant](https://www.vagrantup.com/).
+
+Open a terminal and follow those instructions:
+
+```sh
+# 1. Decide where to put the project. We use "~/Development" in our examples.
+cd ~/Development
+# 2. Clone the vagrant-wagtail-develop repository in a new "wagtail" folder.
+git@github.com:wagtail/vagrant-wagtail-develop.git wagtail
```
Comment (thibaudcolas) [inline README.md]: 👍 done

File: README.md [hunk 5]
```diff
@@ -13,46 +13,133 @@ Features
...
 
-    cd ../wagtail
-    npm run build
+Run the linting:
 
-(or `npm run watch` to watch the source files for changes)
+```sh
+cd /home/vagrant/wagtail
+# Python linting.
+# TODO
```
Comment (thibaudcolas) [inline README.md]: 👍 done

File: README.md (hunk 3 above)
Comment (thibaudcolas) [inline README.md]: 👍 done

File: README.md [hunk 6]
```diff
@@ -129,7 +129,7 @@ cd ~/Development/wagtail/wagtail
 # Change the default origin remote to point to your fork.
 git remote set-url origin https://github.com/<USERNAME>/wagtail.git
 # Add wagtail/wagtail as a "torchbox" remote.
```
Comment (zerolab) [inline README.md]: Maybe change the comment mention to "upstream" too?

Comment (thibaudcolas) [inline README.md]: done :)

**Comment by thibaudcolas** [comment]
I've taken everyone's feedback into account 🚀. To be frank this PR doesn't go as far as I'd like to, but hey it's there already so might as well start with something.

**Comment by gasman** [comment]
Merged in e9e011e59325d20c2fd9f73542d1d4417aed809a - thanks!
